examples/01-multiple-data-sources/backend_b.py

```diff
@@ -49,15 +49,16 @@ class BackendB(DiffSync):
     def load(self):
         """Initialize the BackendB Object by loading some site, device and interfaces from DATA."""
         for site_name, site_data in DATA.items():
-            site = self.site(name=site_name)
-            self.add(site)
+            site, _ = self.get_or_create(self.site, {"name": site_name})
 
             for device_name, device_data in site_data.items():
-                device = self.device(name=device_name, role=device_data["role"], site_name=site_name)
-                self.add(device)
+                identifiers = {"name": device_name}
+                attrs = {"role": device_data["role"], "site_name": site_name}
+                device, _ = self.get_or_create(self.device, identifiers, attrs)
                 site.add_child(device)
 
                 for intf_name, desc in device_data["interfaces"].items():
-                    intf = self.interface(name=intf_name, device_name=device_name, description=desc)
-                    self.add(intf)
+                    identifiers = {"name": intf_name, "device_name": device_name}
+                    attrs = {"description": desc}
+                    intf, _ = self.get_or_create(self.interface, identifiers, attrs)
```

Comment:
I added an example to my latest commit. Now that I can see the example, would it be best to change these to actually add the objects as well rather than having the backend have to check after each call to these new methods?

Reply:
Spoke with Damien and he agreed we should add the new models to the DiffSync adapter. I just pushed up some updates and renaming per Damien (had the old names of the functions)